FlowCal/excel_ui.py

```diff
@@ -357,57 +360,78 @@ def process_beads_table(beads_table,
-            beads_sample_gated = FlowCal.gate.high_low(beads_sample_gated,
-                                                       channels=sc_channels)
+            # Remove saturating events in forward/side scatter, if the FCS data
+            # type is integer. The value of a saturating event is taken
+            # automatically from `beads_sample_gated.range`.
+            if beads_sample_gated.data_type == 'I':
+                beads_sample_gated = FlowCal.gate.high_low(
+                    beads_sample_gated,
+                    channels=sc_channels)
             # Density gating
+            # Axis scaling will be logarithmic if data was acquired with a log
+            # amplifier, and logicle otherwise.
+            sc_amp_type = beads_sample_gated.amplification_type(sc_channels)
             beads_sample_gated, __, gate_contour = FlowCal.gate.density2d(
                 data=beads_sample_gated,
                 channels=sc_channels,
                 gate_fraction=beads_row['Gate Fraction'],
-                xscale='log'
-                       if beads_sample_gated.amplification_type(
-                           sc_channels[0])[0]
-                       else 'linear',
-                yscale='log'
-                       if beads_sample_gated.amplification_type(
-                           sc_channels[1])[0]
-                       else 'linear',
+                xscale='log' if sc_amp_type[0][0] else 'logicle',
+                yscale='log' if sc_amp_type[1][0] else 'logicle',
                 full_output=True)
 
             # Plot forward/side scatter density plot and fluorescence histograms
             if plot:
                 if verbose:
                     print("Plotting density plot and histogram...")
-                # Define density plot parameters
+                # Density plot parameters
                 density_params = {}
                 density_params['mode'] = 'scatter'
-                if beads_sample_gated.amplification_type(sc_channels[0])[0]:
-                    density_params['xscale'] = 'log'
-                else:
-                    density_params['xscale'] = 'linear'
-                if beads_sample_gated.amplification_type(sc_channels[1])[0]:
-                    density_params['yscale'] = 'log'
-                else:
-                    density_params['yscale'] = 'linear'
                 density_params["title"] = "{} ({:.1f}% retained)".format(
                     beads_id,
                     beads_sample_gated.shape[0] * 100. / beads_sample.shape[0])
+                # Axis scaling will be logarithmic if data was acquired with a
+                # log amplifier, and logicle otherwise.
+                sc_amp_type = beads_sample_gated.amplification_type(sc_channels)
+                density_params['xscale'] = \
+                    'log' if sc_amp_type[0][0] else 'logicle'
+                density_params['yscale'] = \
+                    'log' if sc_amp_type[1][0] else 'logicle'
```

Comment:
same as mentioned elsewhere: I think amplifier scale and plotting (and gating) scale are orthogonal, so I would still go with logicle gating, even with a log amplifier.